cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/backup/S3StorageDriver.java

```diff
@@ -97,12 +97,37 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) th
 
             @Override
             public FileVisitResult visitFile(Path path, BasicFileAttributes attrs) throws IOException {
+                int numberOfTriesLeft = 3;
+                while (true) {
+                    try {
+                        return sendOneFile(path, numberOfTriesLeft);
+                    } catch (Exception e) {
+                        numberOfTriesLeft--;
+                        if (numberOfTriesLeft == 0) {
+                            LOGGER.error("Failed uploading file {}", path);
+                            throw new IOException("Failed uploading file" + path);
+                        }
+                        LOGGER.info("Retry visitFile... ");
+                    }
+
+                    try {
+                        Thread.sleep(1);
```

Comment:
This is a sleep of 1ms.  Probably 1000ms is desired.  Please factor out the sleep constant.  I'm sure a sleep for poll rate already exists.  A consistent constant would be gooe.